src/hooks/usePortflioPerformanceData/utils/getApplicablePrice.ts

```diff
@@ -0,0 +1,28 @@
+import { HistoricalPriceData } from '@/types/historicalPrices';
+import { FX_RATE } from '@/utils/portfolioCalculations/const/FX_RATE';
+
+export const getApplicablePrice = (
+  symbolPriceData: HistoricalPriceData | undefined,
+  date: string,
+  lastAvailablePrices: Record<string, number>,
+): number | undefined => {
+  if (!symbolPriceData) return undefined;
```

Comment:
This condition can be avoided if you don't even call this function when `symbolPriceData` is `undefined`.